sites/main-site/src/content/blog/2024/2024-07-02-new-website-structure.mdx

```diff
@@ -0,0 +1,110 @@
+
+<Image
+    src={subsites}
+    class="d-block m-auto mb-2"
+    width={400}
+    density={[1.5, 2]}
+    alt="Three headed dragon meme cartoon, with the title 'nf-co.re sub-sites'. The left head looks angry and has the title 'pipelines', the second head has the title 'docs' and looks stern at the third head, which looks very goofy and has the title 'pipeline-results'."
+/>
+
+# Results: build fast
+
+```mermaid
+---
+config:
+    xyChart:
+        width: 900
+        height: 600
+    themeVariables:
+        xyChart:
+            plotColorPalette: '#1a9655'
+---
+xychart-beta
+    x-axis "sites" ["old", "sites/configs", "sites/docs", "sites/main-sites", "sites/pipelines", "sites/pipelin-releases"]
+    y-axis "average build times [in minutes]" 0 --> 15
+    bar [12, 0.85, 1.5, 1.85, 4.3, 1.5]
+```
+
+# Technical setup
+
+We are using npm workspaces to handle the dependencies in our monorepo structure.
+We tried pnpm which should be even faster and better optimized for this task, but never got it to run
+compeletly, while npm workspaces worked out of the box.
+Every sub-site is a complete Astro project with its own `astro.config.mjs` and `package.json`,
+but the components and layouts are shared.
+We accomplished this by keeping them in `sites/main-site/src` and setting aliases for the relative links
+to `sites/main-site` in the respective sub-sites. This kept the actual needed code changes to a minimum.
+Even though each of these sub-sites are their own entity on our host Netlify, they are all reachable under
+the same `nf-co.re` domain. This is achieved by setting up a redirect rules with a `200` redirect status,
+which is a silent redirect, meaning the URL in the browser does not change.
+
+Data flow for a request to the homepage:
+
+```mermaid
+flowchart LR
+%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
+    A[fas:fa-link https://nf-co.re] --> B(fas:fa-server sites/main-site)
+```
+
+Data flow between sub-sites for a request to a pipeline page:
+
+```mermaid
+flowchart LR
+%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
+
+        E(fas:fa-server sites/main-site)  --fetching components--> D(fas:fa-server sites/pipelines)
+
+    C[fas:fa-link https://nf-co.re/rnaseq/latest/docs/usage] --> B(fas:fa-server sites/main-site)  --redirects silently to --> D
+```
+
+# Changes to the development workflow
+
+All npm commands now should specify the workspace they are run with.
+Instead of running the development server with `npm run dev{:bash}` you need to now run it for a specific sub-site, for example:
```

Comment:
```suggestion
Instead of running the development server with `npm run dev{:bash}` you need to now run it for a specific sub-site.

For example, to see the changes to a blog entry, you should now run:
```